smart_campus/app/views.py

```diff
@@ -295,10 +296,103 @@ def get_all_stations(request):
             'content': station.content,
             'category': str(station.category),
             'location': station.location.get_coords(),
-            'image': [{'image_url': 'http://{0}/{1}'.format(request.get_host(), img.image.url), 'primary': img.is_primary}
-                      for img in station.stationimage_set.all()]
+            'image': {
+                'primary': 'http://{0}{1}'.format(request.get_host(), station.primary_image_url),
+                'others': ['http://{0}/{1}'.format(request.get_host(), img.image.url)
+                           for img in StationImage.objects.filter(station=station, is_primary=False)]
+            }
         }
         for station in Station.objects.all()
     ]
 
     return JsonResponse({'status': 'true', 'message': 'Success', 'data': data}, json_dumps_params={'ensure_ascii': False}, content_type='application/json; charset=utf-8')
+
+
+@csrf_exempt
+@require_POST
+def get_linked_stations(request):
+    """API for retrieving list of stations linked to the Beacon"""
+    beacon_id = request.POST.get('beacon_id')
+
+    stations = Station.objects.filter(beacon__beacon_id=beacon_id)
+    if not stations:
+        return JsonResponse({'status': 'false', 'message': 'No match', 'data': []})
+    else:
+        data = [station.id for station in stations]
+
+        return JsonResponse({'status': 'true', 'message': 'Success', 'data': data})
+
+
+@csrf_exempt
+@require_POST
+def get_single_question(request):
+    """API requesting a single question"""
+    station_id = request.POST.get('station_id')
+    email = request.POST.get('email')
+
+    station = Station.objects.filter(id=station_id).first()
+    user = User.objects.filter(email=email).first()
+
+    if user is None or station is None:
+        return JsonResponse({'status': 'false', 'code': 1, 'message': 'station or user not exist'})
+
+    questions = Question.objects.filter(linked_station=station)
+    user_answered_questions = Question.objects.filter(user=user)
+
+    # Get a question that the user hasn't answered yet
+    question = questions.difference(user_answered_questions).first()
```

Comment:
Do you think that randomly choose one would be better?

Reply:
Sure, I'll give it a try.

Reply:
Maybe you can try method like `random.sample.`